packages/vscode-ui5-language-assistant/src/configuration.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  ConfigurationChangeEvent,
+  Event,
+  WorkspaceConfiguration,
+} from "vscode";
+import { CHANGE_LOG_LEVEL_REQUEST } from "@ui5-language-assistant/language-server";
+
+export const LOGGING_LEVEL_CONFIG_PROP = "UI5LanguageAssistant.logging.level";
+
+export function listenToLogLevelChanges<R>(opts: {
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any -- signature with `any` originates from VSCode APIs */
+  subscriptions: { dispose(): any }[];
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any -- signature with `any` originates from VSCode APIs */
+  sendRequest: (method: string, param: any) => Promise<R>;
+  onDidChangeConfiguration: Event<ConfigurationChangeEvent>;
+  getConfiguration: (section?: string) => WorkspaceConfiguration;
+}): void {
+  opts.subscriptions.push(
+    opts.onDidChangeConfiguration((e) => {
```

Comment:
I wonder if this is necessary. We listen to the `onDidChangeConfiguration` event in the server, doesn't it cover this?

Reply:
it is not 😢 there goes my examples for DI in VSCode ext to avoid sinon mocking